### ISSUE TITLE
Clementine audio output crashes when using PulseAudio sink on remote server

### ISSUE THREAD

**Issue by Clementine-Issue-Importer** [clementine-player/Clementine#4023]
_From [shashank...@gmail.com](https://code.google.com/u/102287654925816727058/) on December 11, 2013 21:10:18_

What steps will reproduce the problem? 1. Take two machines connected over a LAN, (both running ubuntu 13.10 in my case) 
2. Setup pulseaudio to accept remote sources on one machine, to broadcast on the other, share the cookie, so on (not multicast)
3. On machine running clementine, run "pax11publish -e -S <speakerhost>"
4. Run clementine, play music. What is the expected output? What do you see instead? Music should play on the remote host. It does if you do the exact same thing with Amarok 2. Instead, after a few seconds, there is crackling as if buffering, long gaps in playback, and finally it dies entirely. Clementine then needs to be exited and restarted, and again you get a few seconds of sound before it dies again. This period of sound _may_ be related to be pulseaudio buffer length, but I'm not certain. What version of the product are you using? On what operating system? v1.2.1 on Ubuntu 13.10 Please provide any additional information below. I have no idea what information to provide or how. I'm happy to provide specific files, logs, etc. if you can tell me where they can be found.

_Original issue: http://code.google.com/p/clementine-player/issues/detail?id=4023_

**Comment by horazont** [comment]
Indeed, I can confirm this with Clementine 1.2.3 and Pulseaudio 6.0 on Fedora 22 and a remote Pulseaudio 5.0 on Debian 8. I’m willing to put some time into debugging this, if anyone has pointers what to look out for.

**Comment by horazont** [comment]
I actually figured out the problem, possibly.

To verify this, make a simple test. Try to play a video (not just audio, the problem with mpv only manifests with video+audio) using mpv and the remote sink. If it doesn’t work properly (no video playing, but sound works or things like this), we might be having the same cause.

Check the output of ``pacmd list-sinks | grep latency`` on the *local* machine (not the remote). If there are ``current latency`` values way above 1000 ms, you are having the same issue, which is what I think is a bad implementation of module-tunnel-{sink,source}.

However, rescue is on the way: there is module-tunnel-{sink,source}-new. Using these fixes the issue both for clementine and for mpv. To use these you either have to load them manually and statically or patch module-zeroconf-discover to use them instead of module-tunnel-{sink,source}.

It boils down to: this does not seem to be a clementine-specific problem; clementine just tries to be correct with latency and gets shot in the head for this by a bad tunnel implementation, just like mpv does when playing video (in fact, the sink latency continously increases, which is why I assume that mpv playback gets stuck, buffering up more and more audio).

**Comment by fkoester** [comment]
I have the same problem in my setup and I patched my client's PulseAudio so that `module-zeroconf-discover` uses the new tunnel implementation. So far it is working quite well, thanks very much for the hint!

That's the patch:
```patch
Index: pulseaudio-8.0/src/modules/module-zeroconf-discover.c
===================================================================
--- pulseaudio-8.0.orig/src/modules/module-zeroconf-discover.c
+++ pulseaudio-8.0/src/modules/module-zeroconf-discover.c
@@ -214,7 +214,7 @@ static void resolver_cb(
             (a->data.ipv6.address[1] & 0xc0) == 0x80)
             if_suffix = pa_sprintf_malloc("%%%d", interface);
 
-        module_name = pa_sprintf_malloc("module-tunnel-%s", t);
+        module_name = pa_sprintf_malloc("module-tunnel-%s-new", t);
         args = pa_sprintf_malloc("server=[%s%s]:%u "
                                  "%s=%s "
                                  "format=%s "
```
